### PR TITLE
Testsuite buildout

### PR DESCRIPTION
**Pull request by ericfell**

### CODE REVIEW

File: src/rfbzero/experiment.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 from abc import ABC, abstractmethod
 from enum import Enum
+# from typing import Tuple
```
Comment (jeremyfell) [inline src/rfbzero/experiment.py:7]: Do we need this?

File: src/rfbzero/experiment.py [hunk 2]
```diff
@@ -261,6 +333,31 @@ def cycle_step(self) -> CycleStatus:
...
+    charge : bool
+        True if charging, False if discharging.
+    cell_model : ZeroDModel
+        Defined cell parameters for simulating.
+    results: CyclingProtocolResults
+        Container for the simulation result data.
+    update_concentrations: callable
+        Performs coulomb counting, concentration updates via (optional) degradation and crossover mechanisms.
+    current_cutoff: float
+        Current cutoff for CV mode, below it, simulation switches from charge to discharge and vice versa (A).
```
Comment (jeremyfell) [inline src/rfbzero/experiment.py:350]: This line reads a bit weird to me. Maybe `...CV mode. Below it,` instead

File: src/rfbzero/experiment.py [hunk 3]
```diff
@@ -379,11 +480,12 @@ def run(
 
     @staticmethod
     def _validate_cycle_values(
-            value: float,
-            value_charge: float,
-            value_discharge: float,
+            value: float | None,
```
Comment (jeremyfell) [inline src/rfbzero/experiment.py:483]: Note that if we're using union types, our supported Python version will be 3.10+

File: src/rfbzero/experiment.py [hunk 4]
```diff
@@ -426,8 +530,9 @@ def _validate_protocol(
         if cell_model.negative_concentrations():
             raise ValueError('Negative concentration detected')
 
-        def update_concentrations(i: float):
-            return cell_model.coulomb_counter(i, cls_degradation, ncls_degradation, crossover)
+        def update_concentrations(i: float) -> None:
+            """Performs coulomb counting, concentration updates via (optional) degradation and crossover mechanisms"""
```
Comment (jeremyfell) [inline src/rfbzero/experiment.py:534]: Seems odd to have a docstring for a local function. Maybe just have a normal comment?

File: src/rfbzero/experiment.py [hunk 5]
```diff
@@ -720,20 +832,33 @@ def run(
             duration, cell_model, degradation, cls_degradation, ncls_degradation, crossover
         )
 
-        def get_cc_cycle_mode(charge):
+        def get_cc_cycle_mode(charge: bool):
```
Comment (jeremyfell) [inline src/rfbzero/experiment.py:835]: Missing return type annotation? `-> _ConstantCurrentCycleMode`

File: src/rfbzero/experiment.py [hunk 6]
```diff
@@ -720,20 +832,33 @@ def run(
...
                 voltage_limit_capacity_check=False
             )
 
-        def get_cv_cycle_mode(charge, current_estimate, current_lim_cls=None, current_lim_ncls=None):
+        def get_cv_cycle_mode(
+                charge: bool,
+                current_estimate: float,
+                current_lim_cls=None,
+                current_lim_ncls=None
+        ):
```
Comment (jeremyfell) [inline src/rfbzero/experiment.py:851]: Missing return type annotation? `-> _ConstantVoltageCycleMode`

File: src/rfbzero/redox_flow_cell.py [hunk 7]
```diff
@@ -142,12 +159,16 @@ def __init__(self, cls_volume: float, ncls_volume: float, cls_start_c_ox: float,
...
-            raise ValueError("'cls_volume' must be < 'ncls_volume'")
+        if self.init_ocv == 0.0 and self.cls_volume >= self.ncls_volume:
+            raise ValueError("'cls_volume' must be < 'ncls_volume' in a symmetric cell")
 
         if self.init_ocv == 0.0 and self.n_cls != self.n_ncls:
             raise ValueError("Symmetric cell (0 volt OCV) requires 'n_cls' and 'n_ncls' to be equal (same species)")
 
+        if (self.cls_volume * self.n_cls * (self.c_ox_cls + self.c_red_cls)
+                >= self.ncls_volume * self.n_ncls * (self.c_ox_ncls + self.c_red_ncls)):
+            raise ValueError("Initial capacity of CLS must be less than initial capacity of NCLS")
```
Comment (jeremyfell) [inline src/rfbzero/redox_flow_cell.py:170]: Thanks for adding a check. Maybe add two local variables before for init_cls_capacity and init_ncls_capacity to make the if statement more readable

**Review by jeremyfell — COMMENTED** [review]
Thanks, got a few minor comments

File: src/rfbzero/experiment.py (hunk 1 above)
Comment (ericfell) [reply src/rfbzero/experiment.py:7]: nope, removed.

File: src/rfbzero/experiment.py (hunk 2 above)
Comment (ericfell) [reply src/rfbzero/experiment.py:350]: done

File: src/rfbzero/experiment.py (hunk 3 above)
Comment (ericfell) [reply src/rfbzero/experiment.py:483]: >=3.10 is enforced

File: src/rfbzero/experiment.py (hunk 4 above)
Comment (ericfell) [reply src/rfbzero/experiment.py:534]: done

File: src/rfbzero/experiment.py (hunk 5 above)
Comment (ericfell) [reply src/rfbzero/experiment.py:835]: done

File: src/rfbzero/experiment.py (hunk 6 above)
Comment (ericfell) [reply src/rfbzero/experiment.py:851]: done

File: src/rfbzero/redox_flow_cell.py (hunk 7 above)
Comment (ericfell) [reply src/rfbzero/redox_flow_cell.py:170]: done

**Review by jeremyfell — APPROVED** [review]
The new changes look good to me

File: src/rfbzero/experiment.py (hunk 6 above)
Comment (ericfell) [reply src/rfbzero/experiment.py]: thanks for the fix, it's now working

**Pull request merged**